### PR TITLE
End chat when a message is sent to a closed escalated session

### PR DESCRIPTION
**Pull request by bhardwajaditya**
https://github.com/WideChat/Rocket.Chat/issues/886

### CODE REVIEW

File: handlers/LiveAgentSessionHandler.ts [hunk 1]
```diff
@@ -48,4 +56,37 @@ export class LiveAgentSession {
 			console.error(ErrorLogs.LIVEAGENT_SESSION_CLASS_FAILED, error);
 		}
 	}
+
+	private endChat =  async (
```
Comment (Shailesh351) [inline handlers/LiveAgentSessionHandler.ts]: Instead of creating a new function, we can reuse `handleEndChat` from `HandleEndChatCallback.ts`

Comment (Shailesh351) [reply handlers/LiveAgentSessionHandler.ts]: @bhardwajaditya Can you check the above comment?

Comment (bhardwajaditya) [reply handlers/LiveAgentSessionHandler.ts]: We don't have data parameter in this file so can't use handleEndChat as it is. I created this one removing the data requirements

Comment (Shailesh351) [reply handlers/LiveAgentSessionHandler.ts]: @bhardwajaditya In that case, I think we can just create data(ILivechatEventContext) using agent and room and reuse 😄 

```
export interface ILivechatEventContext {
    agent: IUser;
    room: ILivechatRoom;
}
```

File: handlers/LiveAgentSessionHandler.ts [hunk 2]
```diff
@@ -37,7 +39,13 @@ export class LiveAgentSession {
 					messageText = this.message.text;
 				}
 				await sendMessages(this.http, salesforceChatApiEndpoint, persisantAffinity, persistantKey, messageText)
-					.then(() => {
+					.then(async (response) => {
+						if (response.statusCode === 403) {
+							console.error('Send Message: Chat session is expired.', getError(response));
+							console.log(ErrorLogs.LIVEAGENT_SESSION_EXPIRED);
+							this.endChat(assoc, lroom, LcAgent, 'Chat session is expired.');
```
Comment (Shailesh351) [inline handlers/LiveAgentSessionHandler.ts]: Also, I know `lroom` and `LcAgent` were already there with these names. Can you also update the names to `livechaRoom` and `livechatAgent` as you are currently at this file? Thanks in advance 🙂

File: handlers/LiveAgentSessionHandler.ts (hunk 1 above)
Comment (bhardwajaditya) [reply handlers/LiveAgentSessionHandler.ts]: Fixed it

File: handlers/LiveAgentSessionHandler.ts (hunk 2 above)
Comment (bhardwajaditya) [reply handlers/LiveAgentSessionHandler.ts]: Change done

**Comment by ear-dev** [comment]
@Shailesh351 please approve when it looks good.  thanks.

File: handlers/LiveAgentSessionHandler.ts [hunk 3]
```diff
@@ -1,13 +1,17 @@
 import { IHttp, IModify, IPersistence, IRead } from '@rocket.chat/apps-engine/definition/accessors';
 import { IApp } from '@rocket.chat/apps-engine/definition/IApp';
+import { ILivechatEventContext } from '@rocket.chat/apps-engine/definition/livechat';
 import { ILivechatRoom } from '@rocket.chat/apps-engine/definition/livechat/ILivechatRoom';
 import { IMessage } from '@rocket.chat/apps-engine/definition/messages';
+import { RocketChatAssociationRecord } from '@rocket.chat/apps-engine/definition/metadata';
```
Comment (Shailesh351) [inline handlers/LiveAgentSessionHandler.ts]: This import is unused. Can you please remove it? Other things look good to me

**Review by Shailesh351 — APPROVED** [review]
Looks good to me

**Pull request merged**